src/tigerbeetle-java/src/main/java/com/tigerbeetle/Accounts.java

```diff
@@ -0,0 +1,317 @@
+package com.tigerbeetle;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A {@link Batch batch} of accounts.
+ */
+public final class Accounts extends Batch {
+
+    interface Struct {
+
+        public static final int Id = 0;
+        public static final int UserData = 16;
+        public static final int Reserved = 32;
+        public static final int Ledger = 80;
+        public static final int Code = 84;
+        public static final int Flags = 86;
+        public static final int DebitsPending = 88;
+        public static final int DebitsPosted = 96;
+        public static final int CreditsPending = 104;
+        public static final int CreditsPosted = 112;
+        public static final int Timestamp = 120;
+
+        public static final int SIZE = 128;
+    }
+
+    static final Accounts EMPTY = new Accounts(0);
+
+    /**
+     * Constructs an empty batch of accounts with the desired maximum capacity.
+     * <p>
+     * Once created, an instance cannot be resized, however it may contain any number of accounts
+     * between zero and its {@link #getCapacity capacity}.
+     *
+     * @param capacity the maximum capacity.
+     *
+     * @throws IllegalArgumentException if capacity is negative.
+     */
+    public Accounts(final int capacity) {
+        super(capacity, Struct.SIZE);
+    }
+
+    Accounts(final ByteBuffer buffer) {
+        super(buffer, Struct.SIZE);
+    }
+
+    /**
+     * Adds a new account at the end of this batch.
+     * <p>
+     * If successfully, moves the current {@link #setPosition position} to the newly created
+     * account.
+     *
+     * @throws IllegalStateException if this batch is read-only.
+     * @throws IndexOutOfBoundsException if exceeds the batch's capacity.
+     */
+    @Override
+    public void add() {
+        super.add();
+    }
+
+    /**
+     * Gets a unique, client-defined identifier for the account.
+     *
+     * @return an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
```

Comment:
>      * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.

This doesn't make sense to me in context. How is this case possible when the user is just calling a getter?

Reply:
For example 

```java
while (batch.next()) {
  // It is at a valid position
  batch.getId();
}

// It is at an invalid position: "EOF" or "end-of-batch"
// It'll throw an exception here
batch.getId();
```

I think it can use better documentation ...
That is the same thing JDBC's `ResultSet` does if you call `resultSet.getInt("column")` at EOF or before `resultSet.next()`.